planner/mapper/mapper.go

```diff
@@ -844,6 +844,16 @@ func resolveInnerFilterDependencies(
 			// If the key index is outside the bounds of the child mapping array, then
 			// this is not a relation/join and we can add it to the fields and
 			// continue (no child props to process)
+			isDuplicate := false
+			for _, field := range existingFields {
+				if field.GetIndex() == keyIndex {
+					isDuplicate = true
```

Comment:
suggestion: You can drop the `isDuplicate` var by naming the outer loop and continuing here:

```
sourceLoop:
    for key := range source {
         ...
         if field.GetIndex() == keyIndex {
             continue sourceLoop
        }
```

Reply:
yeah, I though about it but the label would be way up there... Anyway I will switch to use the label